fendermint/vm/interpreter/src/fvm/check.rs

```diff
@@ -108,9 +116,23 @@ where
                 } else if self.exec_in_check {
                     // Instead of modifying just the partial state, we will execute the call in earnest.
                     // This is required for fully supporting the Ethereum API "pending" queries, if that's needed.
+                    let (apply_ret, _, latency) =
+                        ElapsedExecution::new(&mut state).execute_explicit(msg.clone())?;
 
-                    // This will stack the effect for subsequent transactions added to the mempool.
-                    let (apply_ret, _) = state.execute_explicit(msg.clone())?;
+                    emit(MsgExecCheck {
+                        height: state.block_height(),
+                        from: msg.from.to_string().as_str(),
+                        to: msg.to.to_string().as_str(),
+                        value: msg.value.to_string().as_str(),
+                        method_num: msg.method_num,
+                        gas_limit: msg.gas_limit,
+                        gas_price: msg.gas_premium.to_string().as_str(),
+                        // TODO Karel - this should be the serialized params
+                        params: msg.params.clone().bytes(),
+                        nonce: msg.sequence,
+                        duration: latency.as_secs_f64(),
```

Comment:
I was thinking of putting the emit into `ElapsedExecution`, so that if no tracking of duration needed, then we just easily replace with `state.execute_explicit`